grudge/execution.py

```diff
@@ -150,25 +149,22 @@ def map_elementwise_max(self, op, field_expr):
     # {{{ nodal reductions
 
     def map_nodal_sum(self, op, field_expr):
-        # FIXME: Could allow array scalars
-        # FIXME: Fix CL-specific-ness
+        actx = self.array_context
         return sum([
-                cl.array.sum(grp_ary).get()[()]
-                for grp_ary in self.rec(field_expr)
-                ])
+            actx.to_numpy(actx.np.sum(grp_ary))[()]
```

Comment:
The `to_numpy` worries me in terms of lazy eval. Can we avoid that?

Reply:
Yep, agreed. Done here: 98e6acf